components/calendar/src/persian.rs

```diff
@@ -0,0 +1,959 @@
+        ];
+
+        let mut leap_year_results: Vec<bool> = Vec::new();
+        for year in leap_year_cycle {
+            let r = Persian::is_leap_year(year);
+            if r {
+                leap_year_results.push(r);
+            }
+        }
+        // 683 is the amount of leap years in a 2820 Persian year cycle
+        assert_eq!(leap_year_results.len(), 683);
+
+        for (index, case) in CASES.iter().enumerate() {
+            leap_years[index] = case.year;
+        }
+        for (year, bool) in leap_years.iter().zip(expected_values.iter()) {
+            assert_eq!(Persian::is_leap_year(*year), *bool);
+        }
+    }
+
+    #[test]
+    fn days_in_provided_year_test() {
+        for case in CASES.iter() {
+            assert_eq!(
+                days_in_provided_year_core(case.year),
+                Persian::days_in_provided_year(case.year)
+            );
+        }
+    }
+
+    #[test]
+    fn test_persian_year_from_fixed() {
+        for (case, f_date) in CASES.iter().zip(FIXED_DATE.iter()) {
+            let date = PersianDateInner(ArithmeticDate {
+                year: (case.year),
+                month: (case.month),
+                day: (case.day),
+                marker: (PhantomData),
+            });
+            assert_eq!(
+                date.0.year as i64,
+                Persian::arithmetic_persian_year_from_fixed(RataDie::new(*f_date)),
+                "{case:?}"
+            );
+        }
+    }
+    #[test]
+    fn test_fixed_from_persian() {
+        for (case, f_date) in CASES.iter().zip(FIXED_DATE.iter()) {
+            let date = PersianDateInner(ArithmeticDate {
+                year: (case.year),
+                month: (case.month),
+                day: (case.day),
+                marker: (PhantomData),
+            });
+
+            assert_eq!(
+                Persian::fixed_from_arithmetic_persian(date).to_i64_date(),
+                *f_date,
+                "{case:?}"
+            );
+        }
+    }
+    #[test]
+    fn test_persian_from_fixed() {
+        for (case, f_date) in CASES.iter().zip(FIXED_DATE.iter()) {
+            let date = Date::try_new_persian_date(case.year, case.month, case.day).unwrap();
+            assert_eq!(
+                Persian::arithmetic_persian_from_fixed(RataDie::new(*f_date)),
+                date,
+                "{case:?}"
+            );
+        }
+    }
+    #[test]
+    fn test_persian_epoch() {
+        let epoch = FIXED_PERSIAN_EPOCH.to_i64_date();
+        // Iso year of Persian Epoch
+        let epoch_year_from_fixed = Iso::iso_from_fixed(RataDie::new(epoch)).inner.0.year;
+        // 622 is the correct ISO year for the Persian Epoch
+        assert_eq!(epoch_year_from_fixed, 622);
+    }
+
+    #[test]
+    fn test_nowruz() {
+        let fixed_date = nowruz(622).to_i64_date();
+        assert_eq!(fixed_date, FIXED_PERSIAN_EPOCH.to_i64_date());
+
+        let years = [
```

Comment:
Convert to Rust range. Define start and end of range as local vars with descriptive names and/or comments.